packages/mobile/src/account/Settings.test.tsx

```diff
@@ -90,4 +99,29 @@ describe('Account', () => {
     )
     expect(tree).toMatchSnapshot()
   })
+
+  it('navigates to PincodeSet screen if entered PIN is correct', async () => {
+    const tree = render(
+      <Provider store={createMockStore({})}>
+        <Settings {...getMockStackScreenProps(Screens.Settings)} />
+      </Provider>
+    )
+    mockedEnsurePincode.mockImplementation(() => true)
```

Comment:
`ensurePincode` doesn't return a boolean, it returns a `Promise<boolean>`, so this test is probably not going to pass :/ You can run individual tests locally using `jest src/account/Settings.test.tsx` or `yarn test src/account/Settings.test.tsx`.

To fix this, instead of `true` here you can return `Promise.resolve(true)` and the same thing for the other test case below. Also, since you have to wait for a promise to resolve it's likely that you'll need to add `jest.runAllTimers()` after firing the press button event, which makes sure that anything async is run.